Create admin file "guidelines-for-spec-writing-and-formatting.md" (#43)
dotnet/csharpstandard: admin/guidelines-for-spec-writing-and-formatting.md

# Guidelines to Spec Writing and Formatting

> 1.	Put definitions of major/often-used terms up front in the Terms-and-definitions clause. Lesser terms should be italicized on their first use only in the narrative.

Reviewer: I believe our discussion about terms-and-definitions ended up with a broad idea that that clause would be more about meta-information about the standard (e.g. what "implementation" means) rather than about specific C# ideas.

> 1.	Don’t say the same thing normatively in more than one place; say it once and point to that place, as needed.

Reviewer: ```suggestion
1.	Don't say the same thing normatively in more than one place; say it once and point to that place, as needed.
```

In general, let's make this file ASCII...

1.	For requirements, use *shall* and *shall not*. Inside informative (that is, non-normative) text, **never** use *shall*, use *must* instead.
1.	Examples and Notes marked as such are informative, and requirements *must* be clear *without* them. Ask yourself, "Does the normative text standalone if all the examples and notes are removed?"
1.	While examples are good, keep them small, simple, and narrowly focused without using unnecessary features. A spec is *not* a tutorial!

> 1.	Avoid using quoted text, as in "natural "ype." If that is a term, except for its first use it should be in Roman without the quotes. In general, such quote marks dilute the contents, especially if their use suggests, “sort of like this thingie." Rather than use an approximate description in quotes, chose the right term or other English text.

Reviewer 2: typo:

```suggestion
1.	Avoid using quoted text, as in "natural type." If that is a term, except for its first use it should be in Roman without the quotes. In general, such quote marks dilute the contents, especially if their use suggests, “sort of like this thingie." Rather than use an approximate description in quotes, chose the right term or other English text.
```

> 1.	Avoid using quoted text, as in "natural "ype." If that is a term, except for its first use it should be in Roman without the quotes. In general, such quote marks dilute the contents, especially if their use suggests, “sort of like this thingie." Rather than use an approximate description in quotes, chose the right term or other English text.

Reviewer 2: We don't specify fonts in Markdown. Should this just say "normal text style" instead of "Roman font"?

Reviewer 3: Checking a few of my fonts from various foundries "regular" might be correct, I think "normal" might be a Word-ism, you also see "plain" used, and "Roman" does refer to a design style... Take your pick from the first three.

Reviewer: +0.5 for regular, +1 for plain +0 for normal, -1 for Roman

> 1.	Avoid using quoted text, as in "natural type." If that is a term, except for its first use it should be in Roman without the quotes. In general, such quote marks dilute the contents, especially if their use suggests, “sort of like this thingie." Rather than use an approximate description in quotes, chose the right term or other English text.

Reviewer: I'm not sure what Roman means here in terms of Markdown. If it's just plain text, that would be clearer.

Additional suggestions:

- Change the curly quote at the start of "soft of like this thingie" to a straight one
- chose => choose

Reviewer: (I see we've talked about Roman before. This is just another instance.)

> 1.	Use singular instead of plural: For example, don’t say, "Tuple types are declared using the following syntax:"; instead, say, "A tuple type is declared using the following syntax:". For the most part, the plural form is quite clear, but there can be cases where it isn’t, so it’s best to use the singular form.

Reviewer: Three curly apostrophes in this bullet.

> 1.	Consider the case in which we have a section with *no* subsections: ## Section S; some text. Later one or more subsections are added, resulting in ## Section S; some text; ### Subsection S.1; some text; ### Subsection S.2; some text. This is *bad*, as a subsequent reference to S is ambiguous; is it to that whole section or is it to the preamble text at the start of that section? As such, you need to put the preamble text in a new subsection (which we've called "General") at the start of that section. Specifically, a section should contain either subsections or text, but never both.

Reviewer: I'd start this bullet with the final sentence. It's hard to start considering a case when I don't know what the area of concern is.

> 1.	Consider the case in which we have a section with *no* subsections: ## Section S; some text. Later one or more subsections are added, resulting in ## Section S; some text; ### Subsection S.1; some text; ### Subsection S.2; some text. This is *bad*, as a subsequent reference to S is ambiguous; is it to that whole section or is it to the preamble text at the start of that section? As such, you need to put the preamble text in a new subsection (which we've called "General") at the start of that section. Specifically, there should *not* be any text between a section head and its first subsection head.

Reviewer 2: Consider this description instead (it was how you described it to me, and it worked well): 
```suggestion
1.	Consider the case in which we have a section with *no* subsections: ## Section S; some text. Later one or more subsections are added, resulting in ## Section S; some text; ### Subsection S.1; some text; ### Subsection S.2; some text. This is *bad*, as a subsequent reference to S is ambiguous; is it to that whole section or is it to the preamble text at the start of that section? As such, you need to put the preamble text in a new subsection (which we've called "General") at the start of that section. Specifically, a section should contain either subsections or text, but never both.
```

# Guidelines to Spec Writing and Formatting

1.	Put meta-information about the standard (e.g., what "implementation" means) in the Terms-and-definitions clause. C#-specific terms should be italicized on their first use only in the narrative.

> 3.	Don't say the same thing normatively in more than one place; say it once and point to that place, as needed.

Reviewer: I'm fine with either leaving everything as "1." in the Markdown or doing the full numbering, but I'd rather we didn't have numbers that look sequential but aren't quite :)

4.	For requirements, use *shall* and *shall not*. Inside informative (that is, non-normative) text, **never** use *shall*, use *must* instead.
5.	Examples and Notes marked as such are informative, and requirements *must* be clear *without* them. Ask yourself, "Does the normative text standalone if all the examples and notes are removed?"
6.	While examples are good, keep them small, simple, and narrowly focused without using unnecessary features. A spec is *not* a tutorial!

> 7.	Avoid using quoted text, as in "natural type." If that is a term, except for its first use it should be in regular without the quotes. In general, such quote marks dilute the contents, especially if their use suggests, "sort of like this thingie." Rather than use an approximate description in quotes, choose the right term or other English text.

Reviewer: I think "regular" needs a noun to support it here - "regular type" for example.
